scripts/g.extension/g.extension.py

```diff
@@ -2218,7 +2285,18 @@ def resolve_known_host_service(url, name, branch):
         else:
             actual_start = ""
         if "branch" in match["url_end"]:
-            suffix = match["url_end"].format(name=name, branch=branch)
+            # Due to strict rate limits for the github API and lack of implementation
+            # for other hosting services, the default branch is currently only
+            # fetched for gitlab
+            default_branch = (
+                get_default_branch(actual_start)
+                if urlparse(actual_start).netloc == "gitlab.com"
+                else "main"
```

Comment:
Isn't this what get_default_branch is doing?

Reply:
Yes. I moved it into the function in the latest commit.

Reply:
As for the "fork" issue, I agree that it should be more clear.
The case I had in mind was an addon-dev contributing a new addon, for example. Until the fork-flag got implemented, it was not possible to test contributions to the official addon repo without cloning the repo of the addon-dev and installing from local URL. So, the main intention has been to allow to install directly from e.g. a feature branch of a (soft) fork of the official repo...
Should be probably added to the manual....

Reply:
> As for the "fork" issue...The case I had in mind was an addon-dev contributing a new addon...

Just for the record, I think in the past (Subversion times) it was possible to install from repositories which behaved like the official repo (I think @neteler or @landam created some). Today, the "test from contributor's fork" use case you mentioned is definitively more important, esp. since individual modules can be installed from small repos (without a need for mirroring the official repo infrastructure).